client/src/utils/context/appParams.utils.ts

```diff
@@ -0,0 +1,261 @@
+/*!
+ * Copyright 2023 - Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { DashboardMessageParams } from "../../features/dashboard/message/DashboardMessage.types";
+import type { HomepageParams } from "../../landing/anonymousHome.types";
+import type { CoreApiVersionedUrlConfig } from "../helpers/url";
+import { DEFAULT_APP_PARAMS } from "./appParams.constants";
+import type {
+  AppParams,
+  AppParamsBooleans,
+  AppParamsStrings,
+  PreviewThresholdParams,
+  PrivacyBannerLayoutParams,
+  TemplatesParams,
+  UploadThresholdParams,
+} from "./appParams.types";
+
+export function validateAppParams(params: unknown): AppParams {
```

Comment:
`validateAppParams` makes me think that the function will just check if the AppParams are valid, but in this case it returns an AppParams object that has been cleaned.

```suggestion
export function validatedAppParams(params: unknown): AppParams {
```